core/dnsserver/register.go

```diff
@@ -14,8 +14,10 @@ import (
 
 const serverType = "dns"
 
+// Any flags defined here, need to be namespaced to the serverType other
+// wise they potentially clash with other server types.
 func init() {
-	flag.StringVar(&Port, "port", DefaultPort, "Default port")
+	flag.StringVar(&Port, serverType+".port", DefaultPort, "Default port")
```

Comment:
Cool, this namespacing would only need to apply for any flags registered by the `dnsserver` package (or any packages/"plugins" it imports, I guess).